libs/contrib/System/Path.idr

```diff
@@ -352,115 +397,147 @@ export
 export
 parent : String -> Maybe String
-parent p = map show $ parent' (parse p)
+parent path = map show $ parent' (parse path)
 
-||| Returns a list of all the parents of the path, longest first,
-||| self included.
+||| Return the list of all parents of the path, longest first, self included.
 |||
 ||| ```idris example
 ||| parents "/etc/kernel" == ["/etc/kernel", "/etc", "/"]
 ||| ```
 export
 parents : String -> List String
-parents p = map show $ iterate parent' (parse p)
+parents path = map show $ List.iterate parent' (parse path)
 
-||| Determines whether base is either one of the parents of full.
+||| Determine whether the base is one of the parents of target.
 |||
 ||| Trailing separator is ignored.
+|||
+||| ```idris example
+||| "/etc" `isBaseOf` "/etc/kernel"
+||| ```
 export
-startWith : (base : String) -> (full : String) -> Bool
-startWith base full = (parse base) `elem` (iterate parent' (parse full))
-
-||| Returns a path that, when appended onto base, yields full.
+isBaseOf : (base : String) -> (target : String) -> Bool
+isBaseOf base target =
+  let 
+    MkPath baseVol baseRoot baseBody _ = parse base
+    MkPath targetVol targetRoot targetBody _ = parse target
+  in
+       baseVol == targetVol
+    && baseRoot == targetRoot
+    && (baseBody `isPrefixOf` targetBody)
+
+||| Return a path that, when appends to base, yields target.
 |||
-||| If base is not a prefix of full (i.e., startWith returns false),
-||| returns Nothing.
+||| Returns Nothing if the base is not a prefix of the target.
 export
-stripPrefix : (base : String) -> (full : String) -> Maybe String
-stripPrefix base full
-    = do let MkPath vol1 root1 body1 _ = parse base
-         let MkPath vol2 root2 body2 trialSep = parse full
-         if vol1 == vol2 && root1 == root2 then Just () else Nothing
-         body <- stripBody body1 body2
-         pure $ show $ MkPath Nothing False body trialSep
+dropBase : (base : String) -> (target : String) -> Maybe String
+dropBase base target =
+  do
+    let MkPath baseVol baseRoot baseBody _ = parse base
+    let MkPath targetVol targetRoot targetBody targetTrialSep = parse target
+    if baseVol == targetVol && baseRoot == targetRoot then Just () else Nothing
+    body <- dropBody baseBody targetBody
+    pure $ show $ MkPath Nothing False body targetTrialSep
   where
-    stripBody : (base : List Body) -> (full : List Body) -> Maybe (List Body)
-    stripBody [] ys = Just ys
-    stripBody xs [] = Nothing
-    stripBody (x::xs) (y::ys) = if x == y then stripBody xs ys else Nothing
+    dropBody : (base : List Body) -> (target : List Body) -> Maybe (List Body)
+    dropBody [] ys = Just ys
+    dropBody xs [] = Nothing
+    dropBody (x::xs) (y::ys) = if x == y then dropBody xs ys else Nothing
 
-||| Returns the final body of the path, if there is one.
-|||
-||| If the path is a normal file, this is the file name. If it's the
-||| path of a directory, this is the directory name.
+||| Returns the last body of the path.
```

Comment:
```suggestion
||| Return the last body of the path.
```

imperative tense, this this change has been carried through the entire file :)

Reply:
ditto